examples/engine/backends.ts

```diff
@@ -20,8 +20,8 @@ export interface SearchIndex {
 
 
 export interface FileStore {
-  async readFile(path: string): string|null;
```

Comment:
Need to know/handle async. Method struct could have Async boolean.

Reply:
i wasn't sure exactly what this should be as a TS definition, because:

<img width="373" alt="image" src="https://github.com/progrium/reflect-ts/assets/5448363/05c7d74d-477e-4f9f-a68b-b49ab25e9452">

Reply:
but also `async` is the same as a function that returns a Promise